src/app.js

```diff
@@ -21,28 +21,96 @@ github.authenticate({
 
 server.get('/', (req, res) => {
   // TODO Return a response that documents the other routes/operations available
+  res.send(`
+    <html>
+      <header><title>Secret Gists!</title></header>
+      <body>
+        <h1>Secret Gists!</h1>
+        <h2>Supported operations:</h2>
+        <ul>
+          <li><i><a href="/gists">GET /gists</a></i>: retrieve a list of gists for the authorized user (including private gists)</li>
+          <li><i><a href="/key">GET /key</a></i>: return the secret key used for encryption of secret gists</li>
+          <li><i>GET /secretgist/ID</i>: retrieve and decrypt a given secret gist
+          <li><i>POST /create { name, content }</i>: create a private gist for the authorized user with given name/content</li>
+          <li><i>POST /createsecret { name, content }</i>: create a private and encrypted gist for the authorized user with given name/content</li>
+        </ul>
+        <h3>Create an *unencrypted* gist</h3>
+        <form action="/create" method="post">
+          Name: <input type="text" name="name"><br>
+          Content:<br><textarea name="content" cols="80" rows="10"></textarea><br>
+          <input type="submit" value="Submit">
+        </form>
+        <h3>Create an *encrypted* gist</h3>
+        <form action="/createsecret" method="post">
+          Name: <input type="text" name="name"><br>
+          Content:<br><textarea name="content" cols="80" rows="10"></textarea><br>
+          <input type="submit" value="Submit">
+        </form>
+      </body>
+    </html>
+  `);
 });
 
 server.get('/gists', (req, res) => {
   // TODO Retrieve a list of all gists for the currently authed user
+  github.gists.getForUser({ username })
+    .then(response => {
+    res.json(response.data);
+  })
+  .catch((err) => {
+    res.json(err);
+  });
 });
 
 server.get('/key', (req, res) => {
   // TODO Return the secret key used for encryption of secret gists
+  res.send(nacl.util.endcodeBase64(key));
 });
 
 server.get('/secretgist/:id', (req, res) => {
   // TODO Retrieve and decrypt the secret gist corresponding to the given ID
+  const id = req.params.id;
+  github.gists.get({id}).then((response) => {
+    const gist = response.data;
+    const filename = Object.keys(gist.files)[0];
+    const blob = gist.files[filename].content;
+    const nonce = nacl.util.decodeBase64(blob.slice(0, 32));
+    const ciphertext = nacl.util.decodeBase64(blob.slice(32, blob.length));
+    const plaintext = nacl.secretbox.open(ciphertext, nonce, key);
+    res.send(nacl.util.encodeUTF8(plaintext));
+  })
```

Comment:
It'll work this way, but even if just for consistency, don't forget your semicolon on 81!